veggieseasons/lib/screens/details.dart

```diff
@@ -125,33 +120,38 @@ class DetailsScreen extends StatelessWidget {
       child: Column(
         crossAxisAlignment: CrossAxisAlignment.stretch,
         children: [
+          Wrap(
+            children: Season.values.map((s) {
+              return SeasonCircle(s, veggie.seasons.contains(s));
+            }).toList(),
+          ),
+          SizedBox(height: 8.0),
           Row(
-            mainAxisSize: MainAxisSize.max,
+            mainAxisSize: MainAxisSize.min,
             children: [
-              Wrap(
-                children: Season.values.map((s) {
-                  return SeasonCircle(s, veggie.seasons.contains(s));
-                }).toList(),
+              CupertinoSwitch(
+                value: veggie.isFavorite,
+                onChanged: (value) {
+                  model.setFavorite(id, value);
```

Comment:
I didn't really follow the pattern. I assume the setState is called somewhere by the mechanism?

Reply:
DetailsScreen fetches a reference to the ScopedModel in its `build` method. When it does that, the ancestor ScopedModel<AppState> (see main.dart) adds its context to the list of listeners. Later, when the CupertinoSwitch's callback calls setFavorite:

https://github.com/flutter/samples/pull/40/files#diff-4b27677843161454d24707f4c9668a26L38

The call to `notifyListeners` marks all the listeners (in this case the DetailsScreen itself) as needing rebuild, which rebuilds the CupertinoSwitch with the new value.

Now that I'm looking at this again, I'm realizing that having two distinct models for app state and settings is an artifact of the way this app was built (two bursts, several months apart), rather than a good design choice. I'm going to need to merge them in a future PR.